bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/IdentityTokenSupplier.java

```diff
@@ -41,6 +42,23 @@ public IdentityTokenSupplier(String audience) {
     lastFetchTime = 0;
   }
 
+  @Override
+  public int hashCode() {
+    return Objects.hashCode(cachedToken, lastFetchTime);
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof IdentityTokenSupplier)) {
+      return false;
+    }
+    IdentityTokenSupplier that = (IdentityTokenSupplier) o;
+    return this.cachedToken.equals(that.cachedToken) && this.lastFetchTime == that.lastFetchTime;
```

Comment:
`return Objects.equal(this.cachedToken, that.cachedToken) && this.lastFetchTime == that.lastFetchTime;` handles potential nulls better

Reply:
Fixed